lib/cisco_node_utils/bfd_global.rb

```diff
@@ -0,0 +1,121 @@
+#
+# NXAPI implementation of BFD Global class
+#
+# May 2016, Sai Chintalapudi
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # node_utils class for bfd_global
+  class BfdGlobal < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail ArgumentError unless name.to_s == 'default'
+      @name = name.downcase
+      set_args_keys_default
+
+      Feature.bfd_enable if instantiate
+    end
+
+    def to_s
+      "bfd_global #{name}"
+    end
+
+    def self.globals
+      hash = {}
+      hash['default'] = BfdGlobal.new('default', false) if
+        Feature.bfd_enabled?
+      hash
+    end
+
+    # Reset everything back to default
```

Comment:
Add `return unless Feature.bfd_enabled?` at the beginning of the `destroy` method.